Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Upsert.ts

```diff
@@ -62,6 +70,16 @@ export async function upsertDocument(
     Logger.debug(`postgresql.repository.Upsert.upsertDocument: Upserting document id ${id}`, traceId);
     await client.query(documentUpsertSql);
 
+    // Delete existing values from the existence table
+    await client.query(deleteExistenceIdsByDocumentId(id));
+
+    // Perform insert of existence ids
+    await client.query(existenceInsertSql(id, id));
+    if (documentInfo.superclassInfo != null) {
+      const existenceId = documentIdForSuperclassInfo(documentInfo.superclassInfo);
+      await client.query(existenceInsertSql(id, existenceId));
+    }
+
```

Comment:
Just to confirm - these multiple queries are in the same transaction context, right? So if line 77 fails, line 74 rolls back?

Reply:
We should test that. Can do a select for update command on the final upsert record, so that there is a lock and the upsert fails. And then ensure that everything else failed.

Reply:
Same transaction context, the transaction starts on line 39 after the try and is ether commited on line 94 or if error rolledback on line 98 in the catch